### PR TITLE
fix(updater): parse git status paths from untrimmed output

### PR DESCRIPTION
**Pull request by BigBrown10**
Addresses the primary abort in #3048: `apply` reporting a false `SAFETY VIOLATION: User file was modified` on a pre-existing dirty user-layer file (e.g. `interview-prep/story-bank.md`), which blocks every update on installs that have an uncommitted file before the updater runs.

## Root cause

`gitStatusEntries()` read status through `git()` → `gitIn()`, which `.trim()`s the whole buffer. `git status --porcelain` emits worktree-modified entries as ` M path` — with a leading space — and when that line is the FIRST line of the buffer, the trim rewrote it into `M path`. The parser then slices `path: line.slice(3)`, which drops the first character: ` M interview-prep/story-bank.md` parsed as `nterview-prep/story-bank.md`.

`initialStatusPaths` (the Set captured by `apply()` before any checkout) is built from those parsed paths, so the guard `if (initialStatusPaths.has(file)) continue;` never matched the real path — `apply()` treated a pre-existing user edit as if the update had touched it and aborted. A hidden-dir path on the first line was mangled the same way (`.antigravitycli/…/SKILL.md` → `antigravitycli/…`).

## Fix

`gitStatusEntries()` now reads via `gitRawIn()` (untrimmed output), keeping the leading space on the first porcelain line so `code: line.slice(0, 2)` and `path: line.slice(3)` land on the exact bytes git emitted. The function is exported with a `root` parameter (default `ROOT`) so the parse can be driven against throwaway repos in tests through the same seam production uses.

## Tests

New `tests/updater-status-paths.test.mjs` (7 cases, all passing) pins the whitespace-sensitive parsing on the exact corruption shapes, including the #3048 repro: a dirty `interview-prep/story-bank.md` as the only pre-existing change, asserting the `initialStatusPaths` guard's `has()` now matches the full path. Cases cover first-line worktree mod (` M`), staged-first-line (`M `), hidden-dir leading-dot path, mixed worktree/staged/untracked entries, clean tree, and Windows CRLF robustness.

Full suite: `node test-all.mjs --quick` → **4316 passed, 0 failed** (4 expected warnings).

## Scope

Addresses the false-positive `SAFETY VIOLATION` abort (the issue title's symptom). Two other symptoms described in #3048 — the whole-set "system files differ from upstream" comparison on a clean checkout, and the rollback `pathspec did not match` errors for net-new upstream files — are separate and not addressed here; they'd be better tracked as their own issues/linked PRs.

Related but intentionally untouched: porcelain paths containing a space arrive C-quoted (`"my notes.md"`) and the parser keeps the quotes — a pre-existing, distinct limitation from the leading-space trim bug under test.

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved update status detection for files in hidden directories and repositories with mixed staged and unstaged changes.
  * Prevented file paths from being misread when Git output contains leading spaces or trailing carriage returns.
  * Preserved accurate handling of pre-existing local changes, helping avoid incorrect update warnings or actions.

* **Tests**
  * Added coverage for clean repositories, varied Git status combinations, and edge-case path parsing.

### CODE REVIEW

File: tests/updater-status-paths.test.mjs [hunk 1]
```diff
@@ -0,0 +1,226 @@
...
+  const { dir, g } = makeRepo();
+  try {
+    writeFileSync(join(dir, 'b.mjs'), 'x\n');
+    g('add', '-A');
+    g('commit', '-qm', 'base');
+    writeFileSync(join(dir, 'b.mjs'), 'y\n');
+
+    const entries = gitStatusEntries(dir);
+    if (entries.length === 1 && entries[0].path === 'b.mjs' && entries[0].code === ' M') {
+      pass('parsed paths carry no trailing carriage return');
```
Comment (Scott-Emberson) [inline tests/updater-status-paths.test.mjs]: Solid suite, and the fix is well proven: I swapped `gitRawIn` back to the old `gitIn` locally and 5 of these 7 fail, so this is a real regression test, not a tautology. One nit on this case: it is titled CRLF / Windows line-ending robustness, but it writes LF files and never puts a CR into the porcelain buffer, so it only passes because git emits LF. The parser (`line.slice(3)` over `split('\n')`) does not strip `\r`, so the robustness the name promises is not actually exercised. Either rename it to something like "porcelain output is LF-delimited", or make it real by running a synthetic `\r\n`-terminated buffer through the split/slice and asserting no trailing `\r`. Non-blocking, since tests 1/3/4/7 already carry the proof.

Comment (BigBrown10) [reply tests/updater-status-paths.test.mjs]: Fair catch, thanks. I made it real instead of renaming it. The parsing now lives in parsePorcelainStatus(), which strips a trailing \r from each line before slicing the path (git can't put a CR inside a path). Test 6 feeds a literal CRLF buffer (' M b.mjs\r\n?? run-now.mjs\r\n') straight through the parser and asserts 'b.mjs' comes back with no trailing CR. I also ran the old slice-on-\n-only logic against the new assertion and it fails, so the test is genuinely driving the fix now. Green locally: updater-status-paths 8/8 and updater-commit-file-modes both pass.

**Comment by santifer** [comment]
Welcome to career-ops, @BigBrown10! This one matters more than its size: gitStatusEntries feeds the did-we-touch-a-user-file safety check, and a path mangled by trim() is a violation that slips past the abort. The byte-for-byte test suite you shipped with it is what keeps this class closed. Merged — your first contribution here 🚀
